### PR TITLE
[ci][release-automation] Use Buildkite commit and make Ray version & commit args for sanity check

### PR DESCRIPTION
**Pull request by khluu**
- Add Ray version & commit hash as input for sanity check script.
- Use these args in all places that call `python sanity_check.py`
- Use `BUILDKITE_COMMIT` instead of `RAY_HASH`
- Remove condition where the block step to download & validate wheels from TestPyPI has to wait on uploading wheels to TestPyPI. This gives more flexibility while running automation.

### CODE REVIEW

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ run_sanity_check() {
         "ray[cpp]==$RAY_VERSION"
     (
         cd release/util
-        python sanity_check.py
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:60]: why is this `BUILDKTIE_COMMIT` but the linux one `RAY_HASH`?

**Review by aslonnie — APPROVED** [review]

File: .buildkite/release-automation/verify-macos-wheels.sh (hunk 1 above)
Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh:60]: the linux script is used for other places other than Buildkite (e.g. calling `pip_download_test.sh`) so calling it `BUILDKITE_COMMIT` doesn't make sense here I think. That step `export RAY_HASH=BUILDKITE_COMMIT` so it's still the same `BUILDKITE_COMMIT` when run on buildkite, but user can modify the hash when they run the script manually.

Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh:60]: fixed

Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh:60]: it's now reading `BUILDKITE_COMMIT` straight from the script without exporting

**Pull request merged**